src/algorithms/time_evolution/simpleupdate3site.jl

```diff
@@ -0,0 +1,465 @@
+
+Note that when `Pa[n] Pb[n]` is identity on a dual space, 
+a twist should be applied to put it to the bond. 
+
+The canonical form is then defined by
+```
+    -←-M̃[n]-←- = -←-Pb[n-1]---M[n]-*-Pa[n]-←-
+        ↓                      ↓
+```
+`-*-` means a twist should be applied if the codomain of `Pa[n]` is a dual space. 
+
+Note that
+```
+    M̃[n]
+    = 1/√s[n-1]←-U†[n-1](R[n-1]--M[n])-*-L[n] V[n]←-1/√s[n]
+    = 1/√s[n-1]←-U†[n-1] Qa[n] (R[n]-*-L[n]) V[n]←-1/√s[n]
+    = 1/√s[n-1]←-U†[n-1] Qa[n] U[n]←-s[n]←-(V†[n] V[n])←-1/√s[n]
+    = 1/√s[n-1]←-U†[n-1] Qa[n] U[n]←-√s[n]
+```
+Then `M̃[n]` (n = 1, ..., N - 1) satisfies the (generalized) left-orthogonal condition
+```
+    ┌---←--M̃[n]--←-     ┌-←- 2
+    |       |           |       
+    s[n-1]  ↓       =   s[n]    (s[0] = 1)
+    |       |           |
+    └---→--M̃†[n]-→-     └-→- 1
+```
+Similarly, we can express M̃ using Qb
+```
+    M̃[n]
+    = 1/√s[n-1]←-U†[n-1] R[n-1]--(M[n]-*-L[n]) V[n]←-1/√s[n]
+    = 1/√s[n-1]←-U†[n-1] (R[n-1]--L[n-1]) Qb[n] V[n]←-1/√s[n]
+    = -*-1/√s[n-1]←-U†[n-1] (R[n-1]-*-L[n-1]) Qb[n] V[n]←-1/√s[n]
+    = -*-1/√s[n-1]←-(U†[n-1] U[n-1])←-s[n-1]←-V†[n-1] * Qb[n] V[n]←-1/√s[n]
+    = -*-√s[n-1]←-V†[n-1] Qb[n] V[n]←-1/√s[n]
+```
+Here `-*-` is a twist to be applied when the codomain of `L[n-1]` is a dual space. 
+Then `M̃[n]` (n = 2, ..., N) satisfies the (generalized) right-orthogonal condition
+```
+    -←-M̃[n]-←┐         1 -←-┐
+        ↓    |              |       
+        *    s[n]   =     s[n-1]   (s[N] = 1)
+        ↓    |              |
+    -→M̃†[n]-→┘         2 -→-┘
+```
+Here `-*-` is the twist on the physical axis. 
+
+# Truncation of a bond on OBC-MPS
+
+Suppose we want to truncate the bond between 
+the n-th and the (n+1)-th sites such that the truncated state
+```
+    |ψ̃⟩  =  M[1]---...---M̃[n]---M̃[n+1]---...---M[N]
+            ↓            ↓      ↓              ↓
+```
+maximizes the fidelity
+```
+            ⟨ψ|ψ̃⟩ ⟨ψ̃|ψ⟩
+    F(ψ̃) = -------------
+            ⟨ψ̃|ψ̃⟩ ⟨ψ|ψ⟩
+```
+This is simply done by using a truncated `Pa[n], Pb[n]`; then
+```
+            M[1]--...-M̃[n]--M̃[n+1]--...-M[N]
+    ⟨ψ|ψ̃⟩ = |         |     |           |
+            M†[1]-...-M†[n]-M†[n+1]-...-M†[N]
+
+            ┌----M̃[n]--M̃[n+1]--┐
+        = s[n-1] |     |     s[n+1]
+            └----M†[n]-M†[n+1]-┘
+
+            ┌--s̃[n]--┐
+        =   |        |  =  s[n].s̃[n]
+            └--s[n]--┘
+```
+Then the fidelity is just
+```
+    F(ψ̃) = (norm(s̃[n], 2) / norm(s[n], 2))^2
+```
+=#
+"""
+Perform QR decomposition through a PEPS tensor
+```
+             ╱           ╱
+    --R0----M---  →  ---Q--*-R1--
+          ╱ |         ╱ |
+```
+"""
+function qr_through(
+    R0::AbstractTensorMap{T,S,1,1}, M::AbstractTensorMap{T,S,1,4}; normalize::Bool=true
+) where {T<:Number,S<:ElementarySpace}
+    @tensor A[-1; -2 -3 -4 -5] := R0[-1; 1] * M[1; -2 -3 -4 -5]
+    q, r = leftorth(A, ((1, 2, 3, 4), (5,)); alg=QRpos())
```

Comment:
```suggestion
    @tensor A[-1 -2 -3 -4; -5] := R0[-1; 1] * M[1; -2 -3 -4 -5]
    q, r = leftorth!(A; alg=QRpos())
```
This might not make a huge difference (yet), but in the long run it's easier to reuse intermediate allocations if they are bundled in `@tensor` calls